src/main/starlark/builtins_bzl/common/java/java_library.bzl

```diff
@@ -164,11 +179,21 @@ JAVA_LIBRARY_ATTRS = merge_attrs(
         "add_exports": attr.string_list(),
         "add_opens": attr.string_list(),
         "licenses": attr.license() if hasattr(attr, "license") else attr.string_list(),
+        # Consumed by _current_repository_transition.
+        "current_repository": attr.string(),
+        "_runfiles_constants": attr.label(
+            default = "@bazel_tools//tools/java/runfiles:java_current_repository",
+            providers = [JavaInfo],
+            cfg = _current_repository_transition,
+        ),
         "_java_toolchain_type": attr.label(default = semantics.JAVA_TOOLCHAIN_TYPE),
+        "_allowlist_function_transition": attr.label(
+            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
+        ),
     },
 )
 
-java_library = rule(
+_java_library = rule(
```

Comment:
Can't change the name here, tooling uses it. Solution is to put `java_library` macro into a new file.